src/compiler/crystal/syntax/lexer.cr

```diff
@@ -1452,14 +1452,14 @@ module Crystal
 
     macro gen_check_int_fits_in_size(type, method, size, actual_type = nil)
       {% if type.stringify.starts_with? "U" %}
-        raise "Invalid negative value #{str} for {{type}}", @token, (current_pos - start) if negative
+        raise "Invalid negative value #{error_number_string} for {{type}}", @token, (current_pos - start) if negative
```

Comment:
I'd suggest refactoring this macro to be _pure_, i.e. avoiding using variables defined in outside scope.

Reply:
Resolved by newest commit. Feels a bit redundant, but probably a good change.